definitions/procedures/puppet/disable_puppet.rb

```diff
@@ -0,0 +1,47 @@
+module Procedures::Puppet
+  class DisablePuppet < ForemanMaintain::Procedure
+    metadata do
+      description 'Disable Puppet feature'
+      confine do
+        feature(:puppet_server)
+      end
+    end
+
+    def run
+      execute!('foreman-rake db:migrate VERSION=0 SCOPE=foreman_puppet') if server_with_puppet?
+      feature(:installer).run(installer_arguments_disabling_puppet.join(' '), :interactive => false)
+      packages_action(:remove, packages_to_remove, :assumeyes => true)
+    end
+
+    private
+
+    def server_with_puppet?
+      find_package(foreman_plugin_name('foreman_puppet'))
+    end
+
+    # TODO: This works only for a katello scenario, we should consider vanilla foreman
+    def installer_arguments_disabling_puppet
+      answers = feature(:installer).answers
+
+      options = []
+      options << '--no-enable-puppet' if answers.key?('puppet')
+      options << '--no-enable-foreman-plugin-puppet' if answers.key?('foreman::plugin::puppet')
+      options << '--no-enable-foreman-cli-puppet' if answers.key?('foreman::cli::puppet')
+      if answers['foreman_proxy']
+        options << '--foreman-proxy-puppet false'
+        options << '--foreman-proxy-puppetca false'
+      end
+      options << '--foreman-proxy-content-puppet false' if answers.key?('foreman_proxy_content')
+      options
+    end
+
+    def packages_to_remove
+      packages = ['puppetserver']
+      if server_with_puppet?
```

Comment:
Open question here to @upadhyeammit: is it safe to simply always remove all packages? I'd think so but not sure.

Reply:
I mean that if you return a package here that isn't installed, will that be properly handled?

Reply:
Yes, it does not fail if package is not installed.